src/transformState.js

```diff
@@ -5,7 +5,36 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (true) {
+      case obj.type === 'addProperties':
+        addProperties(state, obj);
+        break;
+      case obj.type === 'removeProperties':
+        removeProperties(state, obj);
+        break;
+      default:
+        clear(state);
+    };
+  };
```

Comment:
and don't forget to return value at the end of function
```suggestion
  };
  return state;
```